src/CarbonAware.DataSources/CarbonAware.DataSources.ElectricityMaps/test/ElectricityMapsDataSourceTests.cs

```diff
@@ -0,0 +1,103 @@
+using CarbonAware.LocationSources.Exceptions;
+using CarbonAware.DataSources.ElectricityMaps.Client;
+using CarbonAware.DataSources.ElectricityMaps.Model;
+using CarbonAware.Interfaces;
+using CarbonAware.Model;
+using Microsoft.Extensions.Logging;
+using Moq;
+
+namespace CarbonAware.DataSources.ElectricityMaps.Tests;
+
+#pragma warning disable CS8618 // Non-nullable field must contain a non-null value when exiting constructor. Consider declaring as nullable.
+[TestFixture]
+public class ElectricityMapsDataSourceTests
+{
+    private Mock<ILogger<ElectricityMapsDataSource>> _logger { get; set; }
+
+    private Mock<IElectricityMapsClient> _electricityMapsClient { get; set; }
+
+    private ElectricityMapsDataSource _dataSource { get; set; }
+
+    private Mock<ILocationSource> _locationSource { get; set; }
+    private Location _defaultLocation { get; set; }
+    private string _defaultLatitude => _defaultLocation.Latitude.ToString() ?? "";
+    private string _defaultLongitude => _defaultLocation.Longitude.ToString() ?? "";
+    private DateTimeOffset _defaultDataStartTime { get; set; }
+
+    [SetUp]
+    public void Setup()
+    {
+        _logger = new Mock<ILogger<ElectricityMapsDataSource>>();
+        _electricityMapsClient = new Mock<IElectricityMapsClient>();
+        _locationSource = new Mock<ILocationSource>();
+        _dataSource = new ElectricityMapsDataSource(_logger.Object, _electricityMapsClient.Object, _locationSource.Object);
+        _defaultLocation = new Location() { Name = "eastus", Latitude = 34.123m, Longitude = 123.456m };
```

Comment:
Why set this before each test instead of just once when the field is initialized?